server/start.go

```diff
@@ -262,7 +263,18 @@ func startInProcess(ctx *Context, clientCtx client.Context, appCreator types.App
 	if err != nil {
 		return err
 	}
-	tmNode, err := optinode.NewNode(opticonv.GetNodeConfig(cfg), key, proxy.NewLocalClientCreator(app), ctx.Logger)
+	genesis, err := genDocProvider()
+	if err != nil {
+		return err
+	}
+	tmNode, err := optinode.NewNode(
+		context.Background(),
+		opticonv.GetNodeConfig(cfg),
+		key,
```

Comment:
Orthogonal to this change but maybe give the key a slightly more descriptive name? What kind of key is it? It is the key used to identify the node in p2p network right?